src/python/pants/option/parser.py

```diff
@@ -442,7 +442,15 @@ def to_value_type(val_str):
       elif kwargs.get('type') == bool:
         return self._ensure_bool(val_str)
       else:
-        return self._wrap_type(kwargs.get('type', str))(val_str)
+        type_arg = kwargs.get('type', str)
+        if (not kwargs.get('idempotent_type_check', True)) and isinstance(val_str, type_arg):
+          return val_str
+        else:
+          try:
+            return self._wrap_type(type_arg)(val_str)
```

Comment:
Why do we need `idempotent_type_check`? Why not always convert?

Reply:
Because after #7269, enum classes don't have a `.create()` classmethod which accepts instances of the enum as well as valid values for the enum (i.e. the enum constructor will raise if provided an instance of itself). Given the rationale in this file (that all the other types are idempotent, e.g. `str(str(x))` is the same as `str(x)`), the better move here seems to be to remove the `idempotent_type_check` registration option and this conditional here, and to add a check in the `__new__` method for enums to allow passing in an instance of the enum class, or a valid value for the enum.

cc @illicitonion ^ I can't immediately think of a cleaner alternative than the above.

Reply:
Yeah, it does seem reasonable that enum values should be just be idempotent.

Reply:
Done!